### PR TITLE
#298 added sensitivities for logistic model

### PR DESCRIPTION
**Pull request by martinjrobins**
Does this look like a reasonable way to add sensitivities to the models? I've just done the logistic equation for now

### CODE REVIEW

**Comment by MichaelClerx** [comment]
Not sure about this! I think in most cases sensitivities will be calculated at the same time as the ODE solutions, so might be better to have e.g. `simulate(parameters, times, sensitivities=True)`

**Comment by martinjrobins** [comment]
Yea, that sounds better. so return a tuple of `(values,sensitivities)` if `sensitivities=True`?

**Comment by martinjrobins** [comment]
How is this @MichaelClerx?

File: pints/toy/_logistic_model.py [hunk 1]
```diff
@@ -18,10 +18,21 @@ class LogisticModel(pints.ForwardModel):
...
+        f(t) &= \\frac{k}{1+(k/p_0 - 1)*\exp(-r t)} \\\\
+        \\frac{df(t)}{dr} &= \\frac{k t (k / p_0 - 1) \exp(-r t)}
+                              {((k/p_0-1) \exp(-r t) + 1)^2} \\\\
+        \\frac{df(t)}{dk} &= \\frac{k \exp(-r t)}
+                              {p_0 ((k/p_0-1)\exp(-r t) + 1)^2}
+                             + \\frac{1}{(k/p_0 - 1)\exp(-r t) + 1}
+
+    Has two parameters: A growth rate :math:`r` and a carrying capacity
+    :math:`k`. The initial population size :math:`f(0) = p_0` can be set using
+    the (optional) named constructor arg ``initial_population_size``
```
Comment (MichaelClerx) [inline pints/toy/_logistic_model.py]: Nice!

**Comment by MichaelClerx** [comment]
Looks good! Only thing I'm wondering is if it might be better to have a `class ForwardModelWithSensitivities(ForwardModel)` so that we can check the suitability of the model.

However.... Not sure how this will work out in the loglikelihood / error measure etc., so let's wait and see. Maybe add a note for model implementers that sensitivities aren't required?

**Comment by martinjrobins** [comment]
well, at the moment if you try to run `simulate_with_sensitivies` on a model that doesn't have it you get a not implemented exception, which is pretty spot on ;)

**Comment by MichaelClerx** [comment]
@ben18785 @sanmitraghosh Can you have a look and see if this set-up will be able to satisfy your 1st order sensitivity needs?

**Pull request merged**